File: Examples/JavaScript/DemoTest.js
// MathUtils 静态类测试
console.log("Add(3,5):", MathUtils.Add(3, 5));
console.log("Subtract(10,4):", MathUtils.Subtract(10, 4));
console.log("Multiply(6,7):", MathUtils.Multiply(6, 7));
console.log("Divide(8,2):", MathUtils.Divide(8, 2));

// Person 普通类测试
var p = new Person("Alice", 30);
console.log("Person.Name:", p.Name);
console.log("Person.Age:", p.Age);
console.log("Person.Greet():", p.Greet());

// StringHelper 静态类测试
console.log("ToUpper('abc'):", StringHelper.ToUpper("abc"));
console.log("ToLower('ABC'):", StringHelper.ToLower("ABC"));
console.log("IsNullOrEmpty(''):", StringHelper.IsNullOrEmpty(""));
console.log("Reverse('hello'):", StringHelper.Reverse("hello"));

// 错误处理与重试机制测试
try {
    MathUtils.Divide(1, 0);
} catch (e) {
    console.log("捕获异常:", e.message || e);
}

// 异步执行测试（假设支持Promise/async）
(async function() {
    let result = await MathUtils.Add(10, 20);
    console.log("Async Add(10,20):", result);
})();

File: Examples/JavaScript/DemoTest.cs
using System;
using Drx.Sdk.Shared.JavaScript;

namespace Examples.JavaScript
{
    /// <summary>
    /// 单元测试：验证导出类型、错误处理、日志、异步与泛型返回
    /// </summary>
    public class DemoTest
    {
        public static void RunAll()
        {
            var logger = new ScriptLogger { Level = ScriptLogger.LogLevel.Debug };
            logger.Log("=== MathUtils 静态方法测试 ===");
            logger.Log($"Add(2,3): {MathUtils.Add(2, 3)}");
            logger.Log($"Divide(4,2): {MathUtils.Divide(4, 2)}");
            logger.Log($"Divide(1,0): {MathUtils.Divide(1, 0)}");

            logger.Log("=== Person 类测试 ===");
            var p = new Person("Bob", 25);
            logger.Log($"Name: {p.Name}, Age: {p.Age}, Greet: {p.Greet()}");

            logger.Log("=== StringHelper 静态方法测试 ===");
            logger.Log($"ToUpper('abc'): {StringHelper.ToUpper("abc")}");
            logger.Log($"Reverse('hello'): {StringHelper.Reverse("hello")}");

            logger.Log("=== 错误处理与日志 ===");
            try
            {
                MathUtils.Divide(1, 0);
            }
            catch (Exception ex)
            {
                logger.LogException(ex);
            }

            logger.Log("=== 泛型与异步返回测试 ===");
            // 假设有异步接口可用
            // var result = await JavaScript.ExecuteAsync<int>("return MathUtils.Add(5,6);");
            // logger.Log($"Async Add(5,6): {result}");
        }
    }
}